dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -216,6 +246,29 @@ public BinaryRecordReader(InputStream is, long streamPos) throws Exception
         }
     }
 
+    private void startNewRecordBatchSpan()
+    {
+        if (this.recordBuilderSpan != null)
+        {
+            if (this.recordBatchSpan != null)
+            {
+                this.recordBatchSpan.setAttribute("RecordCount", (this.recordCount - this.recordBatchStart));
+                this.recordBatchSpan.setAttribute("DeserializationTimeMS", (this.recordDeserialTimeNS / 1000.0f));
+                this.recordBatchSpan.setAttribute("ExternalProcessingTimeMS", (this.externalProcessingTimeNS / 1000.0f));
+                this.recordBatchSpan.updateName( "RecordBatch[" + this.recordBatchStart + "," + this.recordCount + "]");
+                this.recordBatchSpan.end();
```

Comment:
it's odd to me to end a member span in a method named startNewRecordBatchSpan. Are we closing out a previous span and starting a brand new one?